### PR TITLE
Allow overriding the number of etcd replicas.

### PR DESCRIPTION
**Pull request by mplzik**
This would make etcd clusters resilient to outages of 2 pods while keeping the quorum.

### CODE REVIEW

File: operations/mimir/etcd.libsonnet [hunk 1]
```diff
@@ -2,7 +2,7 @@ local etcd_cluster = import 'etcd-operator/etcd-cluster.libsonnet';
 
 etcd_cluster {
   etcd:
-    $.etcd_cluster('etcd', env=[{
+    $.etcd_cluster('etcd', size=3, env=[{
```
Comment (pstibrany) [inline operations/mimir/etcd.libsonnet:5]: `size=3` doesn't do any increase.

What do you think about adding a new `_config` field to specify the etcd cluster size?

Comment (mplzik) [reply operations/mimir/etcd.libsonnet:5]: Ugh, my bad :-) I'll be happy to add new `_config` entry as well, that might be even easier to manage in the long run.

Comment (mplzik) [reply operations/mimir/etcd.libsonnet]: Fixed; I changed this to be a PR to only allow overriding the default size.

**Review by pstibrany — COMMENTED** [review]
Thanks. Would you please add changelog entry too?

It would be nice to set this value to non-default value in at least one test under `operations/mimir-tests`, and rerun `make build-jsonnet-tests` afterwards to see it working.

**Comment by mplzik** [comment]
@pstibrany I did what you recommended, but it looks like now I'm stuck on some kind of indentation issue. I can fix that locally if needed, but this might be some difference between in tanka formatting between CI and my local installation (I was running latest snapshot)  -- possibly something to look into and fix.

**Comment by pstibrany** [comment]
> @pstibrany I did what you recommended, but it looks like now I'm stuck on some kind of indentation issue. I can fix that locally if needed, but this might be some difference between in tanka formatting between CI and my local installation (I was running latest snapshot) -- possibly something to look into and fix.

I ran `make build-jsonnet-tests` and it produced same whitespace diff. I'll push it to your branch.

File: operations/mimir-tests/test-helm-parity.jsonnet [hunk 2]
```diff
@@ -15,6 +15,7 @@ mimir {
     unregister_ingesters_on_shutdown: false,
     query_sharding_enabled: true,
     overrides_exporter_enabled: true,
+    etcd_replicas: 5,
```
Comment (pstibrany) [inline operations/mimir-tests/test-helm-parity.jsonnet:18]: I'm afraid this file is the wrong for this test :(

`mimir-tests/test-helm-parity-generated.yaml` is used as a base file for testing Helm vs Jsonnet configuration. In Helm we only use 3 etc replicas.

Comment (pstibrany) [reply operations/mimir-tests/test-helm-parity.jsonnet:18]: Perhaps we can make a copy of `test-defaults.jsonnet`, call it `test-etcd-replicas.jsonnet`, and generate yaml file for it? (Test in this case is just inspection that yaml file has 5 replicas specified).

Comment (mplzik) [reply operations/mimir-tests/test-helm-parity.jsonnet:18]: Yup, sure; let me have a stab on that :)

Comment (mplzik) [reply operations/mimir-tests/test-helm-parity.jsonnet]: @pstibrany PTAL.

File: operations/mimir-tests/test-etcd-replicas-generated.yaml [hunk 3]
```diff
@@ -0,0 +1,1305 @@
...
+      value: 1h
+    labels:
+      name: etcd
+    resources:
+      limits:
+        memory: 512Mi
+      requests:
+        cpu: 500m
+        memory: 512Mi
+  size: 5
```
Comment (pstibrany) [inline operations/mimir-tests/test-etcd-replicas-generated.yaml:1304]: 👍

**Review by pstibrany — APPROVED** [review]
Thank you very much!

**Pull request merged**